src/view/table/th.tsx

```diff
@@ -1,153 +1,134 @@
-import { ComponentChild, createRef, h, JSX } from 'preact';
+import { h, ComponentChild, JSX } from 'preact';
 
-import { BaseComponent, BaseProps } from '../base';
 import { classJoin, className } from '../../util/className';
 import { CSSDeclaration, TColumn } from '../../types';
-import { Sort } from '../plugin/sort/sort';
+import { GenericSortConfig, Sort } from '../plugin/sort/sort';
 import { PluginRenderer } from '../../plugin';
-import { JSXInternal } from 'preact/src/jsx';
 import { Resize } from '../plugin/resize/resize';
-
-export interface THProps
-  extends BaseProps,
-    JSX.HTMLAttributes<HTMLTableCellElement> {
-  index: number;
-  column: TColumn;
-  style?: CSSDeclaration;
-}
-
-export interface THState {
-  style: CSSDeclaration;
-}
-
-export class TH extends BaseComponent<THProps, THState> {
-  private sortRef = createRef();
-  private thRef = createRef();
-
-  constructor(props, context) {
-    super(props, context);
-
-    this.state = {
-      style: {},
-    };
-  }
-
-  private isSortable(): boolean {
-    return this.props.column.sort.enabled;
-  }
-
-  private isResizable(): boolean {
-    return this.props.column.resizable;
-  }
-
-  private onClick(e: JSX.TargetedMouseEvent<HTMLInputElement>): void {
-    e.stopPropagation();
-
-    if (this.isSortable()) {
-      this.sortRef.current.changeDirection(e);
-    }
-  }
-
-  private keyDown(e: JSX.TargetedMouseEvent<HTMLInputElement>): void {
-    if (this.isSortable() && e.which === 13) {
-      this.onClick(e);
-    }
-  }
-
-  componentDidMount(): void {
+import { useEffect, useRef, useState } from 'preact/hooks';
+import { useConfig } from '../../hooks/useConfig';
+import * as SortActions from '../plugin/sort/actions';
+import { useStore } from '../../hooks/useStore';
+
+export function TH(
+  props: {
+    index: number;
+    column: TColumn;
+    style?: CSSDeclaration;
+  } & Omit<JSX.HTMLAttributes<HTMLTableCellElement>, 'style'>,
+) {
+  const config = useConfig();
+  const thRef = useRef(null);
+  const [style, setStyle] = useState({});
+  const { dispatch } = useStore();
+
+  useEffect(() => {
     setTimeout(() => {
```

Comment:
Why do we wrap this in a 0 setTimeout? Is it a racing condition?

Reply:
Leftover probably, removed it